components/site/src/lib.rs

```diff
@@ -56,9 +56,12 @@ pub struct Site {
     pub static_path: PathBuf,
     pub templates_path: PathBuf,
     pub taxonomies: Vec<Taxonomy>,
-    /// A map of all .md files (section and pages) and their permalink
+    /// A map of all .md files (sections and pages) and their permalink
     /// We need that if there are relative links in the content that need to be resolved
     pub permalinks: HashMap<String, String>,
+    /// A map of all assets (non .md files colocated to sections and pages) and their permalink, distributed per lang
+    /// We need that if there are relative links in the content that need to be resolved
+    pub assets_permalinks: HashMap<String, HashMap<String, String>>,
```

Comment:
what's the reason for separating by lang? Otherwise we could just add them to the permalinks maps
Are the assets copied for each language? I can't remember

Reply:
Yes, assets are copied for each language.

This will allow to have different permalink depending on the lang for the same asset:
- For the default lang, `path/to/the/asset.jpg` would become `https://yourwebsite.com/slug/to/the/asset.jpg`
- For the non-default lang `lang`, `path/to/the/asset.jpg` would become `https://yourwebsite.com/lang/slug/to/the/asset.jpg`

I tested several things such as always defaulting to the default lang but I find this solution to be more clean and future proof.
When a lang is selected, your assets points to the same base URL than your page.
In addition, even if today it is not the case, if we introduce per-lang assets, it will just require minor adaptations.

Please let me know if you prefer to have something more simple.

Reply:
> Yes, assets are copied for each language.

Ah that's not ideal from content duplication/processing I guess. We can't even define a language specific asset and I don't think it's something that will be added.

I think it makes more sense for assets to be shared across languages and be located once in the folder of the main language page

Reply:
> Ah that's not ideal from content duplication/processing I guess.

This is mandatory and may remain mandatory in the future from my understanding.
We'll break any relative links in non-default language pages otherwise.
Let's imagine we have a `test.file` colocated file and a link defined as follow: `<a href="test.file">`, it would work in default language (resolved as `https://yourwebsite.com/slug/to/the/test.file` by the browser) but would break in non-default ones in case content was not duplicated (resolved as `https://yourwebsite.com/lang/slug/to/the/test.file` by the browser).

> I think it makes more sense for assets to be shared across languages and be located once in the folder of the main language page

If it is still the case, I'll revert my changes to ignore the lang for assets in `get_url` function.
Is it ok for you?

Reply:
Keep like you have right now. In an ideal world, we would serve the assets from a single place and rewrite the links to assets to point to the right thing but that's not for now.